test/unit_tests/braket/ir/ahs/test_ahs_program_v1.py

```diff
@@ -85,3 +86,19 @@ def test__missing_hamiltonian():
     Program(
         setup=valid_setup_input,
     )
+
+
+def test_correct_decimal_serialization():
+    program = Program(
+        setup=valid_setup_input,
+        hamiltonian=valid_hamiltonian_input,
+    )
+    program.hamiltonian.drivingFields[0].amplitude.sequence.times = [Decimal(i * 3e-6 / 7) for i in range(8)]
+    program.hamiltonian.drivingFields[0].amplitude.sequence.values = [Decimal(0.0)] * 8
+
+    serialized_program = program.json()
+    deserialized_program = Program.parse_raw(serialized_program)
+
+    original_ampltiude_times = program.hamiltonian.drivingFields[0].amplitude.sequence.times
+    new_amplitude_times = deserialized_program.hamiltonian.drivingFields[0].amplitude.sequence.times
+    assert new_amplitude_times == original_ampltiude_times
```

Comment:
nit: do you want to check the type in the serialized program?

Reply:
No, I explicitly do not want to check it. I, in fact, don't like serializing with `str`, I'd prefer the behavior of `simplejson.dumps(..., use_decimal=True)`, but I don't know how to code this up, and it would introduce another dependency (simplejson). The purpose of this test is to make sure that serializing and deserializing right after that is leaves the model alone.